app/src/main/java/org/astraea/partitioner/smoothPartitioner/SmoothWeightPartitioner.java

```diff
@@ -126,55 +131,59 @@ public void configure(Map<String, ?> configs) {
       Objects.requireNonNull(mapAddress, "You must configure jmx_servers correctly.");
 
       nodeLoadClient = new NodeLoadClient(mapAddress);
+      dependencyManager = new DependencyManager();
     } catch (IOException e) {
       throw new RuntimeException();
     }
   }
 
   /** Change the weight of the node according to the current Poisson. */
   // visible for test
-  public synchronized void brokerHashMap(Map<Integer, Double> poissonMap) {
+  public synchronized void brokersWeight(Map<Integer, Double> poissonMap) {
     poissonMap.forEach(
         (key, value) -> {
-          if (!brokerHashMap.containsKey(key)) {
-            brokerHashMap.put(key, new int[] {(int) ((1 - value) * 20), 0});
+          var thoughPutAbility = nodeLoadClient.thoughPutComparison(key);
+          if (!brokersWeight.containsKey(key)) {
+            brokersWeight.put(key, new int[] {(int) ((1 - value) * 20 * thoughPutAbility), 0});
           } else {
-            brokerHashMap.put(key, new int[] {(int) ((1 - value) * 20), brokerHashMap.get(key)[1]});
+            brokersWeight.put(
+                key,
+                new int[] {(int) ((1 - value) * 20 * thoughPutAbility), brokersWeight.get(key)[1]});
           }
         });
   }
 
-  private synchronized int allNodesWeight() {
-    return brokerHashMap.values().stream().mapToInt(vs -> vs[0]).sum();
+  private void subBrokerWeight(int brokerID, int subNumber) {
+    brokersWeight.put(
+        brokerID,
+        new int[] {brokersWeight.get(brokerID)[0], brokersWeight.get(brokerID)[1] - subNumber});
   }
 
-  // visible for test
-  public synchronized Map<Integer, int[]> brokerHashMap() {
-    return brokerHashMap;
-  }
-
-  private synchronized void currentBrokerHashMap(Map<Integer, int[]> currentBrokerHashMap) {
-    brokerHashMap = currentBrokerHashMap;
+  private synchronized int allNodesWeight() {
+    return brokersWeight.values().stream().mapToInt(vs -> vs[0]).sum();
   }
 
-  // visible for test
-  public void brokerHashMapValue(Integer x, int y) {
-    brokerHashMap.put(x, new int[] {0, y});
+  public static synchronized DependencyClient dependencyClient(KafkaProducer<?, ?> producer) {
+    return (DependencyClient) Utils.requireField(producer, "partitioner");
```

Comment:
It should handle the casting error if users does not use `SmoothWeightPartitioner`

Reply:
I will resolve this problem in #190,because this question can be put in another related topic.